tremor-script/src/interpreter.rs

```diff
@@ -313,30 +337,42 @@ where
             Gte => Ok(static_bool!(l >= r)),
             Lt => Ok(static_bool!(l < r)),
             Lte => Ok(static_bool!(l <= r)),
-            Add => Ok(Cow::Owned(Value::from(l + r))),
-            Sub => Ok(Cow::Owned(Value::from(l - r))),
-            Mul => Ok(Cow::Owned(Value::from(l * r))),
-            Div => Ok(Cow::Owned(Value::from((l as f64) / (r as f64)))),
-            Mod => Ok(Cow::Owned(Value::from(l % r))),
-            #[allow(clippy::cast_possible_truncation, clippy::cast_sign_loss)]
-            RBitShiftSigned => l.checked_shr(r as u32).map_or_else(
-                || error_invalid_bitshift(outer, inner),
-                |n| Ok(Cow::Owned(Value::from(n))),
-            ),
+            Add => try_math(i64::checked_add, l, r, outer, inner, op),
+            Sub => try_math(i64::checked_sub, l, r, outer, inner, op),
+            Mul => try_math(i64::checked_mul, l, r, outer, inner, op),
+            Div if r != 0 => Ok(Cow::Owned(Value::from((l as f64) / (r as f64)))),
+            Mod if r != 0 => Ok(Cow::Owned(Value::from(l % r))),
+            Mod | Div => error_division_by_zero(outer, inner, op),
+            RBitShiftSigned => u32::try_from(r)
+                .ok()
+                .and_then(|r| l.checked_shr(r))
+                .map_or_else(
+                    || error_invalid_bitshift(outer, inner),
+                    |n| Ok(Cow::Owned(Value::from(n))),
+                ),
+            // This is a bit a bit tricky, for unsigned shift right we need to turnj
+            // our left hand side into a u64 so that the shift right operation does no
+            // preserve signness then turn the result back inot a i64 to match the original
+            // type. We can't use try_from for the u64 <-> i64 conversions
             #[allow(
```

Comment:
are the allows still needed here?

Reply:
yap since we have to do the i64 -> u64 -> i64 cast